.github/workflows/perf.yml

```diff
@@ -0,0 +1,104 @@
+name: Perf CI
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+  workflow_dispatch:
+
+# Checks if any concurrent jobs under the same pull request or branch are being executed
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}-perf
+  cancel-in-progress: true
+
+jobs:
+  perf-libs-linux-amd64:
+    runs-on: [ "self-hosted", "linux", "X64" ]
```

Comment:
We use our own cncf node because:
* we need full hw access for perf counters
* we need hw to stay stable to be able to properly diff perf datas